frontend/hub/collections/hooks/useCopyToRepository.tsx

```diff
@@ -0,0 +1,243 @@
+  collection: CollectionVersionSearch;
+  context: HubContext;
+  onClose: () => void;
+  operation: 'approve' | 'copy';
+}) {
+  const toolbarFilters = useRepositoryFilters();
+  const tableColumns = useRepositoryColumns();
+  const { t } = useTranslation();
+  const { collection } = props;
+  const request = useGetRequest<HubItemsResponse<CollectionVersionSearch>>();
+  const pulpRequest = useGetRequest<PulpItemsResponse<SigningServiceResponse>>();
+
+  const [selectedRepositories, setSelectedRepositories] = useState<Repository[]>([]);
+  const [fixedRepositories, setFixedRepositories] = useState<Repository[]>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [error, setError] = useState<string>('');
+
+  const { collection_auto_sign, require_upload_signatures } = props.context.featureFlags;
+  const autoSign = collection_auto_sign && !require_upload_signatures;
+
+  const operation = props.operation;
+
+  const copyToRepositories = async () => {
+    try {
+      setIsLoading(true);
+
+      const { repository } = props.collection;
+      if (!repository) {
+        return;
+      }
+
+      const pulpId = parsePulpIDFromURL(repository?.pulp_href);
+
+      let signingService = '';
+      const signingServiceName = props.context?.settings?.GALAXY_COLLECTION_SIGNING_SERVICE;
+      if (((operation === 'approve' && autoSign) || operation === 'copy') && signingServiceName) {
+        const url = pulpAPI`/signing-services/?name=${signingServiceName}`;
+        const signingServiceList = await pulpRequest(url);
+        signingService = signingServiceList?.results?.[0].pulp_href;
+      }
+
+      const repoHrefs: string[] = [];
+
+      for (const repo of selectedRepositories) {
+        if (!fixedRepositories.find((item) => item.name == repo.name)) {
+          repoHrefs.push(repo.pulp_href);
+        }
+      }
+
+      const params: {
+        collection_versions: string[];
+        destination_repositories: string[];
+        signing_service?: string;
+      } = {
+        collection_versions: collection.collection_version?.pulp_href
+          ? [collection.collection_version?.pulp_href]
+          : [],
+        destination_repositories: repoHrefs,
+      };
+
+      if (signingService) {
+        params.signing_service = signingService;
+      }
+
+      const api_op = {
+        approve: 'move_collection_version',
+        copy: 'copy_collection_version',
+      }[operation];
+
+      await hubAPIPost(pulpAPI`/repositories/ansible/ansible/${pulpId || ''}/${api_op}/`, params);
+
+      setIsLoading(false);
+      props.onClose();
+    } catch (error) {
+      setError(
+        operation == 'approve' ? t('Can not approve collection.') : t('Can not copy collection')
+      );
+      setIsLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    async function getSelected() {
+      const repos = await request(
+        hubAPI`/v3/plugin/ansible/search/collection-versions?limit=100000&name=${
```

Comment:
limit=10000 will get replaced with limit=100 api side; you should never rely on having the full list available for anything

(there is probably no use for "Select All", "Select page" should be enough)

Reply:
You may understand it wrong. This is not selecting all repositories, this is loading all repositories the collection is in. You think this may lead to problems with performance? If you are ok with it, we can merge it. But after fix - after rebasing, the checkboxes for selection suddenly dissapeared.

Reply:
The checkboxes started to work after reload suddenly again :-D However, I discovered some problem with API - it does not correctly copies collection into some repo, it seems random, but consistent - for example I cant copy to published 2 (it has approved pipeline), the same for old application. So what about merge this and consult this with API and possible there will be API fix, or UI fix on both new and old API.

Reply:
oh, so this is used just to populate the preselected(&disabled) checkboxes?
Feel free to merge, it just won't preselect more than 100 repos, so you may want to add some kind of message like that when there are more.

What's the API problem?

Reply:
We disscussed that in sunday. I probably forgot to create also distribution. Seems not related to this PR.